### PR TITLE
Display ads notifications UI from rewards

### LINKED ISSUE #196: Display custom ads UI in the client when rewards is enabled

### PR DESCRIPTION
**Pull request by kylehickinson**
Fixes https://github.com/brave/brave-rewards-ios/issues/196

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

- Enable rewards by removing NO_REWARDS build flag
- Create a wallet and verify ads is enabled
- Browse a couple websites
- Wait until an ad appears
- Clicking the ad should open the ad URL

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/internal/issues/new?assignees=&labels=security%2C+security-reviews&template=security-review.md&title=iOS:%20Security+Review+for) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -212,6 +213,16 @@ class BrowserViewController: UIViewController {
         contentBlockListDeferred = ContentBlockerHelper.compileBundledLists()
         
         setupRewardsObservers()
+        
+        if let rewards = rewards {
+            notificationsHandler = AdsNotificationHandler(ads: rewards.ads, presentingController: self)
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift:218]: I wonder: will it work well when for example settings or bookmarks panel are open? or AlertViewController?

**Review by iccub — APPROVED** [review]

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (kylehickinson) [reply Client/Frontend/Browser/BrowserViewController.swift:218]: Nope: https://github.com/brave/brave-rewards-ios/blob/1cdecece79e49eca0ce5dc53fad215ca66747858/BraveRewardsUI/Ads/AdsNotificationHandler.swift#L95 :)

Comment (iccub) [reply Client/Frontend/Browser/BrowserViewController.swift:218]: great stuff, hard to find a flaw in your PRs recently :D
pr is approved

Comment (kylehickinson) [reply Client/Frontend/Browser/BrowserViewController.swift:218]: Oh and just so its clear, its not supposed to work when you're on other screens, as per spec: Ads are only shown while the BVC is visible

**Pull request merged**